### PR TITLE
Check if Rosetta installation is necessary

### PR DESCRIPTION
**Pull request by ejortega**
Updates the install script:

- Determine if Rosetta installation needs to proceed by checking if the mac is intel
- Check if Rosetta is already installed and skip Rosetta installation already pr

### CODE REVIEW

File: scripts/setup.sh [hunk 1]
```diff
@@ -26,27 +26,37 @@ validate_internet
...
-	f_echo "Invalid option."
-	exit 1
-	;;
-esac
+# Check if the Mac is Intel or Apple Silicon
+if [[ "$(uname -m)" == "x86_64" ]]; then
+    f_echo "Mac is Intel-based. Rosetta installation is not required."
+else
+	f_echo "- Installation of Rosetta 2 and agreeing to Apple's corresponding software license agreement"
+    if /usr/bin/pgrep oahd >/dev/null 2>&1; then
```
Comment (ichi4096) [inline scripts/setup.sh:34]: This method of checking whether Rosetta is installed or not [is inaccurate.](https://apple.stackexchange.com/a/464165) Change this and I'll aprove your PR.

**Review by ichi4096 — CHANGES_REQUESTED** [review]

File: scripts/setup.sh (hunk 1 above)
Comment (ejortega) [reply scripts/setup.sh]: Updated

**Review by ichi4096 — CHANGES_REQUESTED** [review]
I just tested Your changes and there's a slight hiccup: The "Proceed with installation" prompt only shows up when Rosetta isn't installed. It should however always be shown since it serves to get user consent for all of the above.

**Comment by ejortega** [comment]
> I just tested Your changes and there's a slight hiccup: The "Proceed with installation" prompt only shows up when Rosetta isn't installed. It should however always be shown since it serves to get user consent for all of the above.

I misinterpreted the name then. It appeared to apply only to Rosetta installation because it checks `case $rosetta_install`. However reading the other `f_echo`s it looks to be implied for all terms and not just Rosetta installation.

**Review by ichi4096 — APPROVED** [review]
Nice, everything is in order. Thank You for Your time and repeated contribution.

**Pull request merged**